File: framework/languages/dotnet/src/Zlink.Framework/Runtime/Configuration/Builders/ZLinkStreamCompressionBuilder.cs
namespace Zlink.Framework.Runtime.Configuration.Builders;

internal sealed class ZLinkStreamCompressionBuilder(
    ZLinkFrameworkRegistration registration) : IZLinkStreamCompressionBuilder
{
    public IZLinkStreamCompressionBuilder UseDefault()
    {
        registration.StreamCompressionCodec = ZLinkStreamProtocolDefaults.CreateLz4CompressionCodec();
        return this;
    }

    public IZLinkStreamCompressionBuilder UseLz4()
    {
        return UseDefault();
    }

    public IZLinkStreamCompressionBuilder Use(IZlinkStreamCompressionCodec codec)
    {
        registration.StreamCompressionCodec = codec ?? throw new ArgumentNullException(nameof(codec));
        return this;
    }

    public IZLinkStreamCompressionBuilder Disable()
    {
        registration.StreamCompressionCodec = null;
        return this;
    }
}

File: framework/languages/java/zlink-framework-core/src/main/java/systems/zlink/framework/configuration/ZLinkStreamCompressionBuilder.java
package systems.zlink.framework.configuration;

import systems.zlink.framework.streams.ZLinkStreamCompressionCodec;

public interface ZLinkStreamCompressionBuilder {
    ZLinkStreamCompressionBuilder useDefault();

    ZLinkStreamCompressionBuilder useLz4();

    ZLinkStreamCompressionBuilder use(ZLinkStreamCompressionCodec codec);

    ZLinkStreamCompressionBuilder disable();
}
